src/SprykerSdk/Zed/AopSdk/Business/OpenApi/Builder/OpenApiCodeBuilder.php

```diff
@@ -0,0 +1,675 @@
+     * @param string $path
+     * @param \cebe\openapi\spec\PathItem $pathItem
+     *
+     * @return void
+     */
+    protected function parseApiPathItem(string $path, PathItem $pathItem): void
+    {
+        /** @var \cebe\openapi\spec\Operation $operation */
+        foreach ($pathItem->getOperations() as $method => $operation) {
+            //Set Module Name
+            $this->parsedData[$path][$method]['moduleName'] = $this->getModuleName($path);
+            //Set Controller Name
+            $this->parsedData[$path][$method]['controllerName'] = $this->getController($path);
+
+            if ($operation->requestBody) {
+                //Set Request Body by parsing
+                $this->parseRequestBody($operation, $path, $method);
+            }
+
+            //Set Response Body by parsing
+            $this->parseResponseBody($operation, $path, $method);
+        }
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Operation $operation
+     * @param string $path
+     * @param string $method
+     *
+     * @return void
+     */
+    protected function parseRequestBody(Operation $operation, string $path, string $method): void
+    {
+        /** @var \cebe\openapi\spec\RequestBody $mediaType */
+        foreach ($this->getRequestBodyContent($operation) as $mediaType) {
+            if (isset($mediaType->schema) && $mediaType->schema instanceof Schema) {
+                $this->parsedData[$path][$method]['requestBody'][$this->getPathUsingSchema($mediaType->schema)] = $this->requestBodyParserBySchema($mediaType->schema);
+            }
+            if (isset($mediaType->schema) && $mediaType->schema instanceof Reference) {
+                $this->parsedData[$path][$method]['requestBody'][$this->getPathUsingReference($mediaType->schema)] = $this->requestBodyParserByReference($mediaType->schema);
+            }
+        }
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Operation $operation
+     *
+     * @return iterable
+     */
+    protected function getRequestBodyContent(Operation $operation): iterable
+    {
+        return isset($operation->requestBody) && isset($operation->requestBody->content) ? $operation->requestBody->content : [];
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Schema $schema
+     *
+     * @return iterable
+     */
+    protected function getPropertiesBySchema(Schema $schema): iterable
+    {
+        if (isset($schema->properties)) {
+            return $schema->properties;
+        }
+
+        return [];
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Reference $schema
+     *
+     * @return iterable
+     */
+    protected function getPropertiesByReference(Reference $schema): iterable
+    {
+        if (isset($schema->properties)) {
+            return $schema->properties;
+        }
+
+        return [];
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Schema $schema
+     *
+     * @return array
+     */
+    protected function requestBodyParserBySchema(Schema $schema): array
```

Comment:
The method name is not clear to me.

Reply:
Please extract meaningfull methods. You have lots of foreach, if which can be extracted to make the code more readable.